internal/controller/sync.go

```diff
@@ -260,15 +260,18 @@ func (r *RouteController) buildNextCert(ctx context.Context, route *routev1.Rout
 		return nil, fmt.Errorf("invalid duration annotation on Route %s/%s", route.Namespace, route.Name)
 	}
 
-	var renewBefore time.Duration
+	var renewBefore *metav1.Duration
 	if metav1.HasAnnotation(route.ObjectMeta, cmapi.RenewBeforeAnnotationKey) {
 		renewBeforeAnnotation := route.Annotations[cmapi.RenewBeforeAnnotationKey]
 
-		var err error
-		renewBefore, err = time.ParseDuration(renewBeforeAnnotation)
+		parsedRenewBefore, err := time.ParseDuration(renewBeforeAnnotation)
 		if err != nil {
 			return nil, fmt.Errorf("invalid renew-before annotation %q on Route %s/%s", renewBeforeAnnotation, route.Namespace, route.Name)
 		}
+
+		if parsedRenewBefore > 0 {
```

Comment:
I don't think it is strictly required to check that this `parsedRenewBefore` value is `>0`.
Ultimately, the duration in the annotation should just be passed directly to the Certificate.

The PR fixes the bug by using a pointer to a duration instead of a duration because a 0 duration is also a valid duration. I think we can just focus on that fix.

Reply:
Have updated :+1: